src/main/java/org/isf/menu/manager/UserBrowsingManager.java

```diff
@@ -101,40 +128,43 @@ public User newUser(User user) throws OHServiceException {
 		String username = user.getUserName();
 		if (!username.matches(VALID_USERID_PATTERN)) {
 			throw new OHDataValidationException(
-							new OHExceptionMessage(MessageBundle.getMessage("angal.userbrowser.theusernamecontainsinvalidcharacters.msg")));
+				new OHExceptionMessage(MessageBundle.getMessage("angal.userbrowser.theusernamecontainsinvalidcharacters.msg")));
 		}
-		if (ioOperations.isUserNamePresent(username)) {
+		if (isUserNamePresent(username)) {
 			throw new OHDataIntegrityViolationException(
-							new OHExceptionMessage(MessageBundle.formatMessage("angal.userbrowser.theuseralreadyexists.fmt.msg", username)));
+				new OHExceptionMessage(MessageBundle.formatMessage("angal.userbrowser.theuseralreadyexists.fmt.msg", username)));
 		}
 		return ioOperations.newUser(user);
 	}
 
 	/**
 	 * Updates an existing {@link User} in the DB.
-	 *
 	 * @param user - the {@link User} to update
-	 * @return {@code true} if the user has been updated, {@code false} otherwise.
+	 * @return the {@link User} that has been updated
 	 * @throws OHServiceException When failed to update user
 	 */
-	public boolean updateUser(User user) throws OHServiceException {
+	public User updateUser(User user) throws OHServiceException {
+		User oldUser = ioOperations.getUserByNameAndIsDeleted(user.getUserName());
+
+		if (oldUser != null && user.isDeleted()) {
+			throw new OHServiceException(new OHExceptionMessage("angal.common.denied.msg"));
```

Comment:
```suggestion
			throw new OHServiceException(new OHExceptionMessage(MessageBundle.getMessage("angal.common.denied.msg")));
```

Reply:
Anyway I don't think this kind of message will be clear for the user since it is not even explained why the operation is denied...  if this new key "`angal.common.denied.msg`" is used only in such case when the object is already soft-deleted, I think we could be more explicit, like "`angal.userbrowser.alreadysoftdeleted.msg = The user is already set as deleted.`"

And, for GUI, we could prompt this message even without calling the manager as we already know if the user is already soft-deleted or not.

![image](https://github.com/user-attachments/assets/4b56bb26-0ce8-419c-8fe7-43f08115c870)

![image](https://github.com/user-attachments/assets/240cc99a-5fda-47b7-8a5b-b9d05b8cd72e)